subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/FileSystemWatchingBuildActionRunner.java

```diff
@@ -114,6 +114,11 @@ public boolean isEnabled() {
                 return actuallyWatching;
             }
         });
+        if (action.getStartParameter().isContinuous()) {
```

Comment:
This is probably not part of this spike anyway and you probably already discussed this: but could we implement "manual rebuild" with a keypress (e.g. by clicking r button) for systems where file system watching does not work, and we would just print warning that manual rebuild is required

Reply:
We could, though that is not part of the changes here. Let's see if somebody has use for it.

Currently, I'd say that instead of using a manual trigger, you should run the build again manually - using configuration caching to avoid configuration time.